pkg/collector/corechecks/snmp/internal/checkconfig/config_test.go

```diff
@@ -50,6 +50,7 @@ metrics:
   options:
     placement: 5
     metric_suffix: ReplaceBattery
+    scale_factor: 10
```

Comment:
The `options` section was initially developed for `forced_type` https://github.com/DataDog/integrations-core/pull/7072,
since `scale_factor` is not related, I think it's be better to have it directly at the root:

```
- symbol:
    OID: 1.3.6.1.4.1.318.1.1.1.11.1.1.0
    name: upsBasicStateOutputState
  forced_type: flag_stream
  options:
    placement: 5
    metric_suffix: ReplaceBattery
  scale_factor: 10
```

----

To make it easier to use `sendMetric` be can possibly pass directly a `checkconfig.MetricsConfig` object.

```
func (ms *MetricSender) sendMetric(metricName string, value valuestore.ResultValue, tags []string, metricConfig checkconfig.MetricsConfig) {
```

instead of passing `forcedType string`,  `options checkconfig.MetricsConfigOption` and `scaleFactor`.

Reply:
I can see having historic context can have `options` strictly linked to `forced_type`, thinking outside the box and without that context I see no strict bound between the two, and judged `options` a good fit for this new option. 

Looking at the code I could not see the link as `metric_suffix` doesn't seem to be related to `forced_type`, so thought it was there for options to add to the metric config, and `scale_factor` looks like an option.

The advantage of using `options` is that we prevent from adding one more prop to the `metricConfig`, which has already 7 different options. 

Do you see advantages in having it at the metricsConfig root ?

Reply:
Yes, agree, it's not obvious `options` strictly linked to `forced_type` without context, maybe `forced_type_options` is a better name for `options`.

> Do you see advantages in having it at the metricsConfig root ?

I think currently `options` are meant to be options for a specific config for example `forced_type: flag_stream`. I think it makes sense to have `scale_factor` at metricsConfig since it's a metric config itself and not an option of another config.


====

A somehow related topic. Currently, configs like `metricConfig.forced_type` and `metricConfig.scale_factor` apply to the whole `metricConfig`. It's an issue for table configs since user might want to have a different `forced_type` or `scale_factor` per symbol. It currently leading to some duplication of metric_tags in table configs since user need to redeclare the whole table config with different `forced_type` but same metric_tags.

So, we might want to move some configs to ["symbol" level](https://github.com/DataDog/datadog-agent/blob/ef8bddf28e042d00a53d9b5d122de6c4e1fd1535/pkg/collector/corechecks/snmp/internal/checkconfig/config_metric.go#L18-L28), the only issue is that, "symbol" might not be a good name. Probably need to more thought on the best way to address this.

Reply:
Do you think we should fix this contextually with this PR ? Or would be ok to move on with this feature, which has no breaking changes, and do a refactoring of metric configuration later ? I'm scared if we start working on the refactoring now, we might need deeper changes - would impact metric submission code, metrics stores, and probably more.

I am not sure about what is the priority, if moving `forced_type` to `symbol` has a higher priority than adding `cpu` and `memory` abstract support in SNMP, than we should pause this and consider the refactoring now.

Reply:
> I think currently options are meant to be options for a specific config for example forced_type: flag_stream. I think it makes sense to have scale_factor at metricsConfig since it's a metric config itself and not an option of another config.

Could this be biased by the historical context, and could we consider `options` as generic options ?

Reply:
Will do this change